lib/run_part.c

```diff
@@ -51,68 +52,78 @@ int run_parts(const char *directory, const char *name, const char *action)
 	int n;
 	int execute_result = 0;
 
-	scanlist = scandir(directory, &namelist, 0, alphasort);
+	int dfd = open (directory, O_PATH | O_DIRECTORY | O_CLOEXEC);
+	if (dfd == -1) {
+		fprintf(shadow_logfd, "failed open directory %s: %s\n", directory, strerror(errno));
+		return (1);
+	}
+
+	scanlist = scandir (directory, &namelist, 0, alphasort);
 	if (scanlist<=0) {
+		(void) close (dfd);
 		return (0);
 	}
 
 	for (n=0; n<scanlist; n++) {
-		char         *s;
-		struct stat  sb;
-
-		if (asprintf(&s, "%s/%s", directory, namelist[n]->d_name) == -1) {
-			fprintf(shadow_logfd, "asprintf: %s\n", strerror(errno));
-			for (; n<scanlist; n++) {
-				free(namelist[n]);
+		struct stat sb;
+
+		int fd = openat (dfd, namelist[n]->d_name, O_PATH | O_CLOEXEC);
+		if (fd == -1) {
+			fprintf(shadow_logfd, "failed to open %s/%s: %s\n",
+				directory, namelist[n]->d_name, strerror(errno));
+			for (int k = n; k < scanlist; k++) {
```

Comment:
What does `k` mean?

Reply:
clang-tidy, which I use in my editor, complained about the loop iterator variable `n` being modified inside the loop, so I used a different one.

Reply:
I think it would be more idiomatic to use `i`, no?  That's the usual loop counter.  Or do you have a preference for k for this specific case?